File: Source/Java/Expressive/src/main/java/com/bijington/expressive/functions/logical/IfFunction.java
package com.bijington.expressive.functions.logical;

import com.bijington.expressive.exceptions.ExpressiveException;
import com.bijington.expressive.expressions.IExpression;
import com.bijington.expressive.functions.FunctionBase;
import com.bijington.expressive.helpers.Convert;

/**
 * Created by shaun on 13/07/2016.
 */
public class IfFunction extends FunctionBase {
    @Override
    public String getName() {
        return "If";
    }

    @Override
    public Object evaluate(IExpression[] parameters) throws ExpressiveException {
        this.validateParameterCount(parameters, 3, 3);

        Boolean condition = Convert.toBoolean(parameters[0].evaluate(this.getVariables()));

        return condition ? parameters[1].evaluate(this.getVariables()) : parameters[2].evaluate(this.getVariables());
    }
}

File: Source/CSharp/Expressive/Expressive/Functions/Logical/IfFunction.cs
using Expressive.Expressions;
using System;

namespace Expressive.Functions.Logical
{
    internal class IfFunction : FunctionBase
    {
        #region FunctionBase Members

        public override string Name { get { return "If"; } }

        public override object Evaluate(IExpression[] parameters, ExpressiveOptions options)
        {
            this.ValidateParameterCount(parameters, 3, 3);

            bool condition = Convert.ToBoolean(parameters[0].Evaluate(Variables));

            return condition ? parameters[1].Evaluate(Variables) : parameters[2].Evaluate(Variables);
        }

        #endregion
    }
}
